src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2304 @@
+        else:
+            self.parent.vent()
+        self._refreshPressure()
+
+    def _refreshPressure(self):
+        # Position (vacuum state)
+        state = self.parent.get_chamber_state()
+        val = {"vacuum": PRESSURE_PUMPED if state == "vacuum" else PRESSURE_VENTED}
+        self.position._set_value(val, force_write=True)
+
+        # Pressure
+        pressure = self.parent.get_pressure()
+        if pressure != -1:  # -1 is returned when the chamber is vented
+            self.pressure._set_value(pressure, force_write=True)
+            logging.debug("Updated chamber pressure, %s Pa, vacuum state %s.", pressure, val["vacuum"])
+        else:
+            pressure = 100e3  # ambient pressure, Pa
+            self.pressure._set_value(pressure, force_write=True)
+            logging.warning("Couldn't read pressure value, assuming ambient pressure %s.", pressure)
+
+    def terminate(self):
+        self._polling_thread.cancel()
+
+
+class TerminationRequested(Exception):
+    """
+    Generator termination requested.
+    """
+    pass
+
+
+class SEMDataFlow(model.DataFlow):
+    """
+    This is an extension of model.DataFlow. It receives notifications from the
+    detector component once the SEM output is captured. This is the dataflow to
+    which the SEM acquisition streams subscribe.
+    """
+
+    def __init__(self, detector):
+        """
+        detector (model.Detector): the detector that the dataflow corresponds to
+        """
+        model.DataFlow.__init__(self)
+        self._detector = detector
+
+    # start/stop_generate are _never_ called simultaneously (thread-safe)
+    def start_generate(self):
+        self._detector.start_generate()
+
+    def stop_generate(self):
+        self._detector.stop_generate()
+
+
+class Stage(model.Actuator):
+    """
+    This is an extension of the model.Actuator class. It provides functions for
+    moving the TFS stage and updating the position.
+    """
+
+    def __init__(self, name, role, parent, rng=None, **kwargs):
+        if rng is None:
+            rng = {}
+        stage_info = parent.stage_info()
+        if "x" not in rng:
+            rng["x"] = stage_info["range"]["x"]
+        if "y" not in rng:
+            rng["y"] = stage_info["range"]["y"]
+        if "z" not in rng:
+            rng["z"] = stage_info["range"]["z"]
+        if "rx" not in rng:
+            rng["rx"] = stage_info["range"]["T"]
+        if "rz" not in rng:
+            rng["rz"] = stage_info["range"]["R"]
+
+        axes_def = {
+            "x": model.Axis(unit=stage_info["unit"]["x"], range=rng["x"]),
+            "y": model.Axis(unit=stage_info["unit"]["y"], range=rng["y"]),
+            "z": model.Axis(unit=stage_info["unit"]["z"], range=rng["z"]),
+            "rx": model.Axis(unit=stage_info["unit"]["T"], range=rng["rx"]),
+            "rz": model.Axis(unit=stage_info["unit"]["R"], range=rng["rz"]),
+        } # TODO: make these axis and unit arguments consistent
```

Comment:
What is this TODO about? All axes should be in m/rad. If not, they should be converted between autoscript units and SI.